aten/src/ATen/templates/TensorBody.h

```diff
@@ -36,8 +37,27 @@ namespace torch { namespace autograd {
 
 struct Node;
 
+using hooks_dict = c10::OrderedDict<unsigned, std::function<at::Tensor(const at::Tensor&)>>;
+
 }} // namespace torch::autograd
 
+namespace torch { namespace utils { namespace hooks {
+
+/// A handle which provides the capability to remove a hook.
+struct CAFFE2_API RemovableHandle {
+ public:
+  explicit RemovableHandle(const std::shared_ptr<torch::autograd::hooks_dict>& hooks_dict);
+  void remove() const;
+  unsigned id() const;
+
+ private:
+  std::weak_ptr<torch::autograd::hooks_dict> hooks_dict_ref_;
+  unsigned id_;
+  static std::atomic<unsigned> next_id;
```

Comment:
A static variable on RemovableHandle doesn't seem like a very logical place to specify next id. What would make more sense to me is for this to live along side the hook dict itself (which would also allow you to make it non-atomic). Why is it here?